packages/core/src/renderer/components/+namespaces/store.ts

```diff
@@ -210,7 +211,12 @@ export class NamespaceStore extends KubeObjectStore<Namespace, NamespaceApi> {
 
   @action
   async remove(item: Namespace) {
-    await super.remove(item);
+    if (item.isSubnamespace()) {
+      this.dependencies.deleteSubnamespace(item.getName());
```

Comment:
I don't think this belongs here. `removeNamespace` should be an injectable byitself which does this check and then calls `deleteSubnamespace`.

Futhermore, this is a floating promise.

Reply:
Removing a regular namespace made by using store `super.remove()` method. If I move all of the logic above to the separate injectable, I have to call store method again.

Reply:
Fixed floating promise.

Reply:
Calling the store method is not the problem, maybe I am misunderstanding what you are meaning.

I would think that we should have the following:

```
const removeNormalNamespaceInjectable = ...; // This is a wrapper around `NamespaceStore.remove`
const removeSubnamespaceMarkerInjectable = ...; // This is the code that removes the subnamespace marker

const removeNamespaceInjectable = (namespace) => {
  if (namespace.isSubnamespace()) {
    await removeSubnamespaceMarker();
  }

  await removeNormaltNamespace();
};
```